retreaver-typescript/src/Campaign.ts

```
@@ -0,0 +1,398 @@
+   */
+  private handleGoogleAnalyticsIntegration(number: RetreaverNumber): void {
+    const googleAnalyticsConfig: GoogleAnalyticsConfig = this.getIntegrationConfig(number, "google_analytics");
+    if (typeof googleAnalyticsConfig === "undefined") {
+      return;
+    }
+
+    const obtainGoogleAnalyticsCookies = new Promise<{gaSessionId: string; gaClientId: string}>((resolve) => {
+      const googleAnalyticsInterval = setInterval(() => {
+        const gaSessionIdPattern = /_ga_[^=]+=([^;]*)/;
+        const gaSessionMatch = document.cookie.match(gaSessionIdPattern);
+        const gaSessionMatched = gaSessionMatch && gaSessionMatch[1];
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)